include/boost/multiprecision/cpp_int.hpp

```diff
@@ -801,7 +803,8 @@ struct cpp_int_base<MinBits, MinBits, unsigned_magnitude, Checked, void, false>
        : m_wrapper(i),
          m_limbs(1) {}
    BOOST_MP_FORCEINLINE BOOST_MP_CXX14_CONSTEXPR cpp_int_base(signed_limb_type i) noexcept((Checked == unchecked))
-       : m_wrapper(limb_type(i < 0 ? static_cast<limb_type>(-static_cast<signed_double_limb_type>(i)) : i)), m_limbs(1)
+       : m_wrapper(i < 0 ? static_cast<limb_type>(-static_cast<signed_double_limb_type>(i)) : static_cast<limb_type>(i)),
```

Comment:
This would be safer using `unsigned_abs` here as well.

Reply:
Done